sympy/functions/combinatorial/numbers.py

```diff
@@ -1400,6 +1401,181 @@ def _eval_is_positive(self):
         if n.is_nonnegative and n.is_integer:
             return True
 
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                             Motzkin numbers                                #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class motzkin(Function):
+    r"""
+    Motzkin numbers
+
+    Motzkin number is the number of different ways of drawing
+    non-intersecting chords between n points on a circle
+    (not necessarily touching every point by a chord).
+    The numbers satisfy the recurrence relation
+
+    .. math :: M_n = \frac{2n + 1}{n + 2} M_{n - 1} + \frac{3n - 3}{n + 2} M_{n - 2}
+
+    For non-negative integers, it can be calculated as
+
+    .. math :: M_n = \sum_0^{\left\lfloor n/2 \right\rfloor } {n \choose 2k} C_k
+
+    where `C_k` is the `k^{th}` Catalan number.
+
+    It can be generalized to complex numbers as
+
+    .. math :: M_n = {}_2 F_1\Bigg(\frac{\frac{1 - n}{2}, \frac{-1}{2}}{2} \Bigg{|} 4\Bigg)
+
+    where `{}_p F_q` is the Generalized Hypergeometric function.
+
+    * ``motzkin(n)`` gives the `n^{th}` Motzkin number.
+
+    * ``find_first_n_motzkin(n)`` gives the first
+    `n` Motzkin numbers.
+
+    * ``find_motzkin_numbers_in_range(n)`` gives
+    the all Motzkin numbers in the interval `[x, y]`.
+
+    * ``is_motzkin(n)`` returns the position of
+    `n` in the Motzkin sequence of numbers if `n`
+    is a Motzkin number, else returns False
+
+    * ``is_prime(n)`` returns True if `n` is
+    a Motzkin prime number, else returns False
+
+    Examples
+    ========
+
+    >>> from sympy import motzkin
+    >>> motzkin(23)
+    1129760415
+    >>> motzkin(12.132)
+    17673.6827206284 + 0.00229270467798908*I
+
+    >>> motzkin.find_first_n_motzkin(15)
+    [1, 1, 2, 4, 9, 21, 51, 127, 323, 835, 2188, 5798, 15511, 41835, 113634]
+    >>> motzkin.find_motzkin_numbers_in_range(50, 100000)
+    [51, 127, 323, 835, 2188, 5798, 15511, 41835]
+    >>> motzkin.is_motzkin(3192727797)
+    24
+    >>> motzkin.is_motzkin(486431896)
+    False
+    >>> motzkin.is_prime(953467954114363)
+    True
+
+    >>> from sympy import Symbol, Sum, binomial, hyper
+    >>> n = Symbol('n')
+
+    >>> motzkin(n).rewrite(Sum)
+    Sum(binomial(n, 2*_k)*catalan(_k), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(binomial)
+    Sum(binomial(2*_k, _k)*binomial(n, 2*_k)/(_k + 1), (_k, 0, floor(n/2)))
+
+    >>> motzkin(n).rewrite(hyper)
+    hyper((0.5 - 0.5*n, -0.5*n), (2,), 4)
+
+    See Also
+    ========
+
+    bell, bernoulli, binomial, catalan, euler, fibonacci, tribonacci, harmonic, lucas,
+    genocchi, partition, sympy.functions.special.hyper.hyper
```

Comment:
Should we add motzkin to the see alsos of other functions?

Reply:
Also, this seems like a long list. Is motzkin actually related to all of these? I don't think it's necessary to list every combinatorial function here, just the ones that are similar or somehow related to motzkin.

Reply:
> Should we add motzkin to the see alsos of other functions?

I forgot to do this. For what functions do you think I should add it to?

> Also, this seems like a long list. Is motzkin actually related to all of these? I don't think it's necessary to list every combinatorial function here, just the ones that are similar or somehow related to motzkin.

I was just following the current convention which links all numbers to each other. Let me know what to add/remove.